docs/src/architecture/data-model.md

```diff
@@ -0,0 +1,253 @@
+It covers the thin-edge device itself, as well as other _external devices_ connected to thin-edge.
+Each device is represented with a data **object**. 
+Each data **object** can have different **fields**. 
+The following JSON code shows a small example of a device data **object**:
+       
+```json
+          {
+              "name": "thin-edge device",
+              "type": "thin-edge.io"
+          }
+```
+
+**thin-edge** manages an **inventory** to store and provide all those objects. 
+
+The figure below illustrats the **data-model** objects and the **inventory**:
+
+![thin-edge Inventory](images/inventory.svg)
+
+* The **thin-edge Device** object represents the device it-self, that runs **thin-edge** and manages that **inventory**.
+  That objects has the fields `name` and `type` that contains the device-name and device-type visible in the cloud.
+
+* A **Child-Device** object could be exist more than once in the inventory. 
+  Each **Child-Device** object represents an _external device_ (e.g. sensor, actuator, PLC, any other kind of device) that is connected to the thin-edge device.
+  * Each **child-device** object is assocoiated with a separate individual device in the cloud. 
+  * Similar to the **thin-edge Device** object, each **child-device** object has the fields `name` and `type`.
+    In addition, each **child-device** object has a field `childid`, that contains a unique ID to address that child-device.
+  * NOTE: Not just _external devices_, but also processes running on the thin-edge device itself, can be represented with a **child-device** object in the **inventory** - to treat them as __logical child-devices__.
+
+* Each **Capability** object represents a functionality a device is capable.
+  * A capability could be by example _Configuration Management_, _Log file Management_ or _Software Management_, or any custom specific capability provided by a custom specific plugin.
+  * The content and structure of each **capability** object is very specific to the capability it represents. E.g. a **capability** object for _Configuration Management_ contains a list of configuration files the device supports, whereas a **capability** object for _Software Management_ contains details about installable package-types the device supports.
+  * A schema that describes the content and structure of **capability** objects for a certain capability is called **Capability Schema**. More details see section below [Capability Schemas](#capability-schemas).
+  * A device object can contain several **capability** objects.
+
+## Capability Schemas
+
+The content and structure of each **capability** object in the inventory is very specific to the capability it represents. To document and standardize the content and structure of those objects for certain capabilities the **Capability Schemas** are used. That way each **external child-device** can use each **plugin**, as long as both use the same **capability schema**.
```

Comment:
I miss a key point, that a capability schema is defined by the operation plugin. 

* By the way this is related to @itsroshan 's comment about the capability  and operation terms. One support the other and it might good to simplify the vocabulary.
* The term *plugin* is too vague. Operation plugin would be better. I would even prefer "operation software component"